code/modules/projectiles/guns/energy/kinetic_accelerator.dm

```diff
@@ -31,22 +31,62 @@
 	. = ..()
 	if(max_mod_capacity)
 		. += "<b>[get_remaining_mod_capacity()]%</b> mod capacity remaining."
-		. += span_info("You can use a <b>crowbar</b> to remove modules.")
+		. += span_info("You can use a <b>crowbar</b> to remove all modules or <b>right-click</b> with an empty hand to remove a specific one.")
 		for(var/A in modkits)
 			var/obj/item/borg/upgrade/modkit/M = A
 			. += span_notice("There is \a [M] installed, using <b>[M.cost]%</b> capacity.")
 
 /obj/item/gun/energy/recharge/kinetic_accelerator/crowbar_act(mob/living/user, obj/item/I)
 	. = TRUE
 	if(modkits.len)
-		to_chat(user, span_notice("You pry the modifications out."))
+		to_chat(user, span_notice("You pry all the modifications out."))
 		I.play_tool_sound(src, 100)
 		for(var/a in modkits)
 			var/obj/item/borg/upgrade/modkit/M = a
 			M.forceMove(drop_location()) //uninstallation handled in Exited(), or /mob/living/silicon/robot/remove_from_upgrades() for borgs
 	else
 		to_chat(user, span_notice("There are no modifications currently installed."))
 
+/obj/item/gun/energy/recharge/kinetic_accelerator/attack_hand_secondary(mob/user, list/modifiers)
+	. = ..()
+	if(. == SECONDARY_ATTACK_CANCEL_ATTACK_CHAIN)
+		return
+
+	if(!LAZYLEN(modkits))
+		return
+
+	var/list/display_names = list()
+	var/list/items = list()
+	for(var/modkits_lenght in 1 to length(modkits))
+		var/obj/item/thing = modkits[modkits_lenght]
+		display_names["[thing.name] ([modkits_lenght])"] = REF(thing)
+		var/image/item_image = image(icon = thing.icon, icon_state = thing.icon_state)
+		if(length(thing.overlays))
+			item_image.copy_overlays(thing)
+		items += list("[thing.name] ([modkits_lenght])" = item_image)
+
+	var/pick = show_radial_menu(user, src, items, custom_check = CALLBACK(src, PROC_REF(check_menu), user), radius = 36, require_near = TRUE, tooltips = TRUE)
+	if(!pick)
+		return
+
+	var/modkit_reference = display_names[pick]
+	var/obj/item/borg/upgrade/modkit/modkit_to_remove = locate(modkit_reference) in modkits
+	if(!istype(modkit_to_remove))
+		return
+	if(!user.put_in_hands(modkit_to_remove))
+		modkit_to_remove.forceMove(get_turf(src))
```

Comment:
```suggestion
		modkit_to_remove.forceMove(drop_location())
```

Reply:
`drop_location` alone was throwing errors so I checked how other items go around it and it was by declaring(?) a new `drop_location` before using it.

```
user = new(drop_location())			
M.forceMove(drop_location())
```

I have no idea if that is cleaner than `get_turf` so please tell me if I can do it in a better way.

Reply:
I see plenty of working usages of it, what error were you getting?

Reply:
I think it was an issue with undefined procs, I found it weird because there is another use of `drop_location` error on the file that was not throwing the same error.

Removed the line now and it is not throwing error so it was likely just me VSC or me being dumb.